### PR TITLE
Built ClaimedDefinitions to manage claimed types and services on generation

### PR DESCRIPTION
**Pull request by oldergod**

### CODE REVIEW

File: wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt [hunk 1]
```diff
@@ -261,11 +261,12 @@ data class KotlinTarget(
 }
 
 data class ProtoTarget(
-  override val includes: List<String> = listOf(),
-  override val excludes: List<String> = listOf(),
-  override val exclusive: Boolean = true,
   val outDirectory: String
 ) : Target() {
+  override val includes: List<String> = listOf()
```
Comment (Egorand) [inline wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt]: This change is unrelated to `ClaimedDefinitions`, wanna move it to a separate PR?

Comment (oldergod) [reply wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt]: yeah removed it
